### PR TITLE
Barricade

### PR DESCRIPTION
**Pull request by TerryKang**

### CODE REVIEW

**Comment by jacobmcphail** [comment]
Hey make sure add the requested changes asked by Isaac.
https://github.com/bcit-datacomm/4981_Linux_Game/pull/8

**Comment by jacobmcphail** [comment]
Hey Terry, a large merge has happen and you will probably need to update your code to work with it.

File: src/Entity.h [hunk 1]
```diff
@@ -20,21 +20,26 @@ class Entity {
...
 
     float getY() const; // get y coordinate
 	
 	void updateHitBoxes(); // update hitbox positions
 
-    HitBox movementHitBox;	// Hit box for movement
-    HitBox projectileHitBox; // Hit box for projectiles
-    HitBox damageHitBox; // Hit box for damage
+  std::shared_ptr<HitBox> movementHitBox = nullptr;
+	std::shared_ptr<HitBox> projectileHitBox = nullptr;
```
Comment (isaacmorneau) [inline src/Entity.h]: We just got rid of all the tabs in the other PR, We cant have them return again. Please follow the spacing as outlined in the [style guide](https://github.com/bcit-datacomm/4981_Linux_Game/wiki/Style-Guide#dont-use-tabs-ever-period-use-4-spaces)

File: src/Entity.h [hunk 2]
```diff
@@ -20,21 +20,26 @@ class Entity {
...
+	std::shared_ptr<HitBox> pickupHitBox = nullptr;
+
     LTexture texture;
     SDL_Rect spriteClips[1];
 
 private:
-    float x = 200; //x coordinate
-    float y = 200; //y coordinate
+
+    float x = 0; //x coordinate
```
Comment (isaacmorneau) [inline src/Entity.h]: defaults go in the ctor not the member variable section

File: src/Barricade.cpp [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+}
+
+bool Barricade::isPlaceable(){
+	return boolPlaceable;
+}
+
+// Move Zombie by x and y amount
+void Barricade::move(float playerX, float playerY, float moveX, float moveY, CollisionHandler * ch) {
+	this->setPosition(moveX, moveY);
```
Comment (isaacmorneau) [inline src/Barricade.cpp]: `this->` is redundant, please remove

File: src/Barricade.h [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+    bool isPlaceable();
+
+    bool checkPlaceablePosition(float, float, float, float, CollisionHandler*);
+
+private:
+    double angle = 180.0;
+    int health = 100;
+    int state; //used to select sprite to display
+    bool boolPlaceable = false;
```
Comment (isaacmorneau) [inline src/Barricade.h]: defaults go in ctor not in the declaration

File: src/Entity.h [hunk 5]
```diff
@@ -20,22 +20,26 @@ class Entity {
...
 	std::shared_ptr<HitBox> damageHitBox = nullptr;
 	std::shared_ptr<HitBox> pickupHitBox = nullptr;
+
     LTexture texture;
     SDL_Rect spriteClips[1];
 
 private:
+
     float x = 0; //x coordinate
     float y = 0; //y coordinate
```
Comment (isaacmorneau) [inline src/Entity.h]: defaults go in the ctor not in the declaration

File: src/GameManager.cpp [hunk 6]
```diff
@@ -282,9 +286,18 @@ void GameManager::updateCollider() {
         collisionHandler.quadtreeDam->insert(m.second.damageHitBox.get());
     }
 
+   	for (const auto& b : this->barricadeManager) {
+		  this->collisionHandler->quadtreeMov->insert(m.second.movementHitBox.get());
+		  this->collisionHandler->quadtreeDam->insert(m.second.damageHitBox.get());
```
Comment (isaacmorneau) [inline src/GameManager.cpp]: same as above, remove `this->`

File: src/GameManager.h [hunk 7]
```diff
@@ -68,6 +72,8 @@ class GameManager {
     std::map<unsigned int, Zombie> zombieManager;
     std::map<unsigned int, Turret> turretManager;
     std::map<unsigned int, WeaponDrop> weaponDropManager;
+    std::map<unsigned int, Barricade*> barricadeManager;
```
Comment (isaacmorneau) [inline src/GameManager.h]: This needs to hold the objects themselves. not the pointers, please read the guide on [pointer implementation](https://github.com/bcit-datacomm/4981_Linux_Game/wiki/Implementation-Guide#pointer-usage). Please mimic the other managers when implementing this.

File: src/Zombie.h [hunk 8]
```diff
@@ -28,6 +31,7 @@ class Zombie : public Movable {
     Zombie();
     virtual ~Zombie();
 
+    int getRandomAngle(); //randomly generate the angle of zombie
 private:
 
     int health = 100;
```
Comment (isaacmorneau) [inline src/Zombie.h]: defaults in the ctor

**Pull request merged**